ruby/yarp
# Package up YARP as a gem

__Summary__

- Wrap YARP as a gem
  - add a gemspec
  - add a "rake build" task to package the gem
  - add a "rake check_manifest" task to validate the gem packaging
  - extconf.rb knows how to make librubyparser
  - add CI jobs to test gem packaging and installation
- Add test coverage for symbol visibility in static archives and shared objects
- Switch the compile-time default to NOT export symbols on non-windows platforms, superseding #1111.

__Notable change__

The 'compile' rake task is no longer a dependency of the 'test' task, to allow us to run the test suite against the installed gem in CI. CONTRIBUTING.md has been updated. Developers can still run `bundle exec rake` to do all the things.

__Notable design choice__

"autoconf" is run during gem packaging, and the "configure" script is in the gem. This reduces the requirements on the installation system.
